### ISSUE TITLE
Support Kafka 0.9

### ISSUE THREAD

**Issue by acchen97** [logstash-plugins/logstash-output-kafka#32]
As Kafka 0.9 is near release, we should support this new version along with new features like authentication and wire encryption.

**Comment by seang-es** [comment]
Kafka 0.9 supports TLS/SSL (https://issues.apache.org/jira/browse/KAFKA-1684). We've been getting some customer interest in supporting this feature in our input/output plugins.

**Comment by suyograo** [comment]
This is being actively worked on. We are targeting 2.2 for this.

**Comment by ebuildy** [comment]
That's mean is not working with Kafka 0.9 currently?

I tried with Kafka 2.11-0.9 (this one http://mirrors.ircam.fr/pub/apache/kafka/0.9.0.0/kafka_2.11-0.9.0.0.tgz), I am not seeing any errors and any messages as well ;-(

**Comment by talevy** [comment]
@ebuildy, In this case, "supporting 0.9" means to update the producer to depend on the latest package release, which would enable support for SSL. The current release of the plugin is still fully compatible with the wire protocol of kafka 0.9 and should continue to work against a kafka 0.9 cluster (without SSL).

**Comment by ebuildy** [comment]
Thanks you, must be an issue with my setup, so The Truth Is Out There ;-( I can see the topic is created by logstash, but I don't see any message with:

    kafka-console-consumer.sh --zookeeper docker:2181 --topic logstash-raw

My logstash config is pretty simple:

     kafka
        {
            topic_id => "logstash-raw"
            bootstrap_servers => "kafka:9092"
            linger_ms => 5
            timeout_ms => 5
        }

Do I miss something important with Kafka (I am quite new on it), many thanks for your help.

**Comment by ajviradia** [comment]
Will Logstash also support SSL + SASL with Kafka 0.9.0?

**Comment by type0lang** [comment]
+1

**Comment by vaijab** [comment]
> This is being actively worked on. We are targeting 2.2 for this.

@suyograo just wondering where is it being worked on if it still is? I have a kafka 0.9.x deployment and would love to start using new consumer and producer protocol.

**Comment by cptcanuck** [comment]
FWIW, Kafka 0.10 is slated for late April, with breaking protocol changes.  Maybe better to focus on support there, instead of everything for 0.9, just to change it again soon?

**Comment by vaijab** [comment]
@cptcanuck that's a fair point, but there will always be another version of kafka API which breaks backwards compatibility until kafka reaches 1.0, I assume.

I had a quick look and it should be fairly simple to add kafka 0.9.x producer support to logstash-output-kafka. I will give it a go.

**Comment by suyograo** [comment]
@vaijab 0.9 for this output is work in progress here:  https://github.com/logstash-plugins/logstash-output-kafka/pull/36

As for consumer, its in this branch: https://github.com/logstash-plugins/logstash-input-kafka/tree/update/0.9

Idea is to release a beta this week.

**Comment by suyograo** [comment]
Working on adding integration tests

**Comment by vaijab** [comment]
> Idea is to release a beta this week.

@suyograo awesome stuff. I'll definitely give it a go at testing from tomorrow. Thanks.

**Comment by brouwerorange** [comment]
Hi
Thank you for supporting Kafka0.9.
I am using Kafka0.9.0.1 with SASL+Kerberos.
I can't see properties related with SASL+Kerberos.
If there is, please let me know the link. i am glad to test it for this release.
or is there any plan to support it also?

**Comment by suyograo** [comment]
@vaijab Beta 1 versions are out

```
bin/plugin install --version 3.0.0.beta1 logstash-output-kafka
```

@brouwerorange this beta does not have SASL support yet

**Comment by gmoskovicz** [comment]
@suyograo looks like this issue should be closed? Kafka 0.90 support is already there right? 

What i see that is missing is:

1. SSH tests (integration and unit).
2. Some examples in the documentation on using SSH.
3. Fix old documentation to include this settings.
